### ISSUE TITLE
ChildNode.nextSibling should be another ChildNode, not just a Node

### ISSUE THREAD

**Issue by jcormont** [microsoft/TypeScript#28551]
**TypeScript Version:**  3.1.6

**Search Terms:** ChildNode Node nextSibling

**Code**

```ts
// a basic loop over elements in the DOM
let element = document.body.firstChild;  // is a ChildNode | null
while (element) {
    // ... do something with element
    let sibling = element.nextSibling;  // is a Node | null -- why?
    element = sibling;  // uh-uh!
}
```

**Expected behavior:**

Surely a sibling of a child is itself also a child (of the same parent). So siblings should be assignable using the same variable.

**Actual behavior:**

Error:
```
Type 'Node | null' is not assignable to type 'ChildNode | null'.
  Type 'Node' is not assignable to type 'ChildNode'.
    Property 'after' is missing in type 'Node'. [2322]
```

**Related Issues:** 

I think this is what #27453 also refers to, but fails to describe accurately so it has been closed.

In any case this is related to the 'solution' in #24633 which clearly shows the type of nextSibling as Node, not ChildNode.

**Comment by Fabianopb** [comment]
Updating `nextSibling: Node | null` to `nextSibling: ChildNode | null` sounds reasonable.

`ChildNode` extends `Node` (which extends `EventTarget`) so I imagine it won't be a breaking change, just adding the correct type. @saschanaz why do you think `ChildNode ` should be this union type you mentioned?

Btw, I'd be glad to grab this story if you guys haven't started yet!

**Comment by Fabianopb** [comment]
PR opened, maybe you guys want to continue the conversation in there.

**Comment by saschanaz** [comment]
>I wonder what was the reason then to make ChildNode extend Node in that commit. 

I think he just wanted to remove `Node & ChildNode`.

**Comment by Fabianopb** [comment]
https://github.com/Microsoft/TSJS-lib-generator/pull/619 is merged, should this be closed?

**Comment by jakebailey** [comment]
Just looking at ancient issues; yes, the above was merged and then brought over in #29690.
